File: surf-friends/surf-friends-velocity/src/main/java/dev/slne/surf/friends/velocity/command/FriendCommand.java
package dev.slne.surf.friends.velocity.command;

import dev.jorel.commandapi.CommandAPICommand;

import dev.slne.surf.friends.velocity.VelocityInstance;
import dev.slne.surf.friends.velocity.command.subcommand.friend.FriendAcceptCommand;
import dev.slne.surf.friends.velocity.command.subcommand.friend.FriendAddCommand;
import dev.slne.surf.friends.velocity.command.subcommand.friend.FriendDenyCommand;
import dev.slne.surf.friends.velocity.command.subcommand.friend.FriendRemoveCommand;
import dev.slne.surf.friends.velocity.command.subcommand.friend.FriendSaveCommand;
import dev.slne.surf.friends.velocity.command.subcommand.friend.FriendToggleCommand;

public class FriendCommand extends CommandAPICommand {
    public FriendCommand(String name) {
        super(name);
        withPermission("surf-social.friends.friend.command");

        withSubcommand(new FriendAcceptCommand("accept"));
        withSubcommand(new FriendAddCommand("add"));
        withSubcommand(new FriendAddCommand("request"));
        withSubcommand(new FriendDenyCommand("deny"));
        withSubcommand(new FriendRemoveCommand("remove"));
        withSubcommand(new FriendToggleCommand("toggle"));
        withSubcommand(new FriendSaveCommand("save"));

        executesPlayer((player, info) -> {
            VelocityInstance.getInstance().openMenu(player, "friends:main");
        });
    }
}

File: surf-friends/src/main/java/dev/slne/surf/social/friend/command/friend/FriendCommand.kt
package dev.slne.surf.social.friend.command.friend

import dev.jorel.commandapi.CommandAPICommand

class FriendCommand(commandName: String) : CommandAPICommand(commandName) {
    init {
        withPermission("surf.social.friend.command")
        withSubcommand(FriendToggleCommand("toggle"))
        withSubcommand(FriendAcceptCommand("accept"))
        withSubcommands(FriendDenyCommand("deny"))
        withSubcommands(FriendSendCommand("add"))
        withSubcommands(FriendRemoveCommand("remove"))
        withSubcommands(FriendListCommand("list"))
    }
}
